saleor/core/extensions/manager.py

```diff
@@ -0,0 +1,181 @@
+import importlib
+from typing import TYPE_CHECKING, Any, List, Union
+
+from django.conf import settings
+from django_countries.fields import Country
+from prices import Money, MoneyRange, TaxedMoney, TaxedMoneyRange
+
+from ..taxes import TaxType, quantize_price
+
+if TYPE_CHECKING:
+    from ...checkout.models import Checkout, CheckoutLine
+    from ...product.models import Product
+    from ...account.models import Address
+    from ...order.models import OrderLine, Order
+
+
+class BaseManager:
+    """Base manager for handling a plugins logic"""
+
+    plugins = None
+
+    def __init__(self, plugins: List[str]):
+        self.plugins = []
+        for plugin_path in plugins:
+            plugin_path, _, plugin_name = plugin_path.rpartition(".")
+            plugin_module = importlib.import_module(plugin_path)
+            plugin_class = getattr(plugin_module, plugin_name)
+            self.plugins.append(plugin_class())
+
+    def run_plugin_method(self, name: str, default_value: Any, *args, **kwargs):
+        value = default_value
+        for p in self.plugins:
```

Comment:
Is it just plain reduce here?
We could handle it a little more clearly in few ways, with reduce or simpler loop:
```python
methods = (getattr(p, name) for p in self.plugins if hasattr(p, name))
value = default_value
for method in methods:
    value = method(*args, **kwargs, previous_value=value)
return value
```